share/location/osm.py

```diff
@@ -49,21 +63,21 @@ def osm_get_place_hierarchy(place_id : int):
         # check if data already exists, if so, return prematurely as within that location relation
         # the caller has everything it nneeds
         ext_data = {}
-        r = LocationExternal.get_by_custom(LocationExternalSourceEnum.OSM, 'place_id', next_place_id)
-        if len(r) > 0:
-            locations.append(r[0])
+        r = Location.get_by_custom(LocationExternalSourceEnum.OSM, 'place_id', next_place_id)
+        if r != None:
+            #locations.append(r[0])
+            locations.append(r)
             break
-            
-        # build and perform osm query
-        response = requests.get('{}{}'.format(url, next_place_id), timeout=QUERY_TIMEOUT)
-        if response.status_code != 200:
-            e = LookupError('failed request to openstreetmap; status code {}'.format(response.status_code))
-            raise(e)
-        response_json = json.loads(response.text)
+       
+        response_json = osm_get_detail(next_place_id)
+        current_place_id = next_place_id
+        next_place_id = response_json['parent_place_id']
```

Comment:
Considering the `osm_get_detail` method returns a dict if I am not wrong. I think it is safer to use the `.get()` method which should default to `None` in the event that the key you're looking for is absent. Calling this:
```
response_json['parent_place_id']
```
will raise a `keyError` if its absent.
On one hand I imagine its reasonably safe to assume that the value will always be returned in the response but I prefer the `get` just to be safe.

Reply:
I would actually prefer an exception here, as the data coming back should _always_ have this field. If it doesn't, we will catch either wrong assumptions and/or changes in the source data structure.